yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,259 @@
+import functools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    OnDemandPagedList,
+    UnsupportedError,
+    int_or_none,
+    parse_qs,
+    str_or_none,
+    traverse_obj,
+    urljoin,
+)
+
+
+_FIELDS = '''
+    _id
+    clipImageThumb
+    clipLink
+    clipLinkHLS
+    clipTitle
+    createdDate
+    shareId
+    user { _id }
+    username
+    views'''
+
+_EXTRA_FIELDS = '''
+    clipLength
+    clipSizeBytes
+    game'''
+
+_QUERIES = {
+    'clip': '''query ($id: String!) {
+        video: getClip(clipIdentifier: $id) {
+            %s %s
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'montage': '''query ($id: String!) {
+        video: getMontage(clipIdentifier: $id) {
+            %s
+        }
+    }''' % _FIELDS,
+    'Clips': '''query ($page: Int!, $user: String!, $game: Int) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: false, game: $game) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'Montages': '''query ($page: Int!, $user: String!) {
+        videos: montages(search: createdDate, page: $page, user: $user) {
+            data { %s }
+        }
+    }''' % _FIELDS,
+    'Mobile Clips': '''query ($page: Int!, $user: String!) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: true) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+}
+
+
+class AllstarBaseIE(InfoExtractor):
+    @staticmethod
+    def _parse_video_data(video_data):
+        def _media_url_or_none(path):
+            return urljoin('https://media.allstar.gg/', str_or_none(path))
+
+        def _profile_url_or_none(path):
+            return urljoin('https:/allstar.gg/u/', str_or_none(path))
+
+        return traverse_obj(video_data, {
+            'id': ('_id', {str_or_none}),
+            'display_id': ('shareId', {str_or_none}),
+            'title': ('clipTitle', {str_or_none}),
+            'url': ('clipLink', {_media_url_or_none}),
+            'thumbnail': ('clipImageThumb', {_media_url_or_none}),
+            'duration': ('clipLength', {int_or_none}),
+            'filesize': ('clipSizeBytes', {int_or_none}),
+            'timestamp': ('createdDate', {int_or_none}),
+            'uploader': ('username', {str_or_none}),
+            'uploader_id': ('user', '_id', {str_or_none}),
+            'uploader_url': ('user', '_id', {_profile_url_or_none}),
+            'view_count': ('views', {int_or_none}),
+            'categories': ('game', {str_or_none}),
```

Comment:
Categories should be an array. I also don't think `'730'` is a good category. Maybe just remove it?
```suggestion
```

Reply:
yes I think you are right.
done.